backend/src/main/java/capstone/backend/services/ProductService.java

```diff
@@ -36,26 +40,32 @@ public ProductDTO getProductDetails(Long id) {
     }
 
     public ProductDTO createProduct(ProductDTO product) throws EntityWithThisIdAlreadyExistException {
-        if (product.getId() != null && repo
-                .findById(product.getId())
-                .isPresent()) {
+        if (productExists(product)) {
             throw new EntityWithThisIdAlreadyExistException(String.format("Product %s already has the id %d", product.getName(), product.getId()));
         }
         return ProductMapper.mapProductWithDetails(repo.
                 save(ProductMapper.mapProduct(product)));
     }
 
     public ProductDTO editProduct(ProductDTO product) {
-        if (repo
-                .findById(product.getId())
-                .isEmpty()) {
+        if (!productExists(product)) {
             throw new EntityNotFoundException(String.format("Couldn't find a product with the id %d", product.getId()));
         }
         return ProductMapper.mapProductWithDetails(repo
```

Comment:
hard to read with this line break => IntelliJ autoformat?

Reply:
this is autoformatted. I don't get what line break you refer to

Reply:
I wouldn't break after "repo"

Reply:
alright, I see it now. the preview actually cut just on 'repo' therefore I diidn't notice